compiler/damlc/daml-stdlib-src/DA/Text.daml

```diff
@@ -205,14 +205,22 @@ dropWhile p = implode . P.dropWhile p . explode
 dropWhileEnd : (Text -> Bool) -> Text -> Text
 dropWhileEnd p = implode . L.dropWhileEnd p . explode
 
+-- Separate function because recursion is only allowed at the top-level and we
+-- don't want to pay for many implode/explode operations.
+-- This shoud not be exported.
+splitOnHelper : [Text] -> [Text] -> [Text]
+splitOnHelper s [] = []
+splitOnHelper s t | s == t = ["", ""]
+splitOnHelper s t = case L.stripInfix s t of
+                    None -> [implode t]
+                    Some (before, after) -> implode before :: splitOnHelper s after
```

Comment:
I'm generally in favor of making indentation depth independent of identifier lengths:
```suggestion
    None -> [implode t]
    Some (before, after) -> implode before :: splitOnHelper s after
```

Reply:
Fixed.